### PR TITLE
fix: misc component fixes

### PR DESCRIPTION
**Pull request by BrianAmadori**
This PR fix many issues found doing the following:

- Starting explorer using Unity editor and WSS mode
- Start in -20,80 ORG (xmas event scene)
- Walk around the scene borders continuously

By doing this, many errors were found due to badly handled messaging edge cases and specific `LoadableShape` timing issues.

### CODE REVIEW

**Review by pravusjif — APPROVED** [review]

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/LoadableShape.cs [hunk 1]
```diff
@@ -197,43 +196,50 @@ protected virtual void ConfigureColliders(DecentralandEntity entity)
...
+            if (loadWrapper.entity.gameObject == null) return;
 
-            if (entity.meshesInfo.currentShape != null)
-            {
-                entity.meshesInfo.renderers = entity.meshRootGameObject.GetComponentsInChildren<Renderer>();
+            GameObject go = loadWrapper.entity.gameObject;
 
-                var model = (entity.meshesInfo.currentShape as LoadableShape).model;
-                ConfigureVisibility(entity.meshRootGameObject, model.visible, loadWrapper.entity.meshesInfo.renderers);
+            go.name += " - Failed loading";
```
Comment (AjimenezDCL) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/LoadableShapes/LoadableShape.cs:214]: Let's wrap this in a `UNITY_EDITOR` directive.

File: unity-client/Assets/Scripts/MainScripts/DCL/Components/UI/UIShape.cs [hunk 2]
```diff
@@ -42,6 +42,10 @@ public float GetScaledValue(float parentSize)
         {
             if (type == Unit.PIXELS) return value;
 
+            // Prevent division by zero
+            if (parentSize <= Mathf.Epsilon)
+                parentSize = 1;
```
Comment (AjimenezDCL) [inline unity-client/Assets/Scripts/MainScripts/DCL/Components/UI/UIShape.cs:47]: This is an edge case, but wouldnt it be better to set `parentSize` to `Mathf.Epsilon`?

File: unity-client/Assets/UnityGLTF/Scripts/GLTFComponent.cs [hunk 3]
```diff
@@ -336,22 +342,23 @@ public IEnumerator LoadAssetCoroutine()
...
-            loadingRoutine = null;
-            Destroy(loadingPlaceholder);
-            Destroy(this);
         }
 
+
         private bool TestDistance()
         {
-            if (mainCamera == null)
+            if (mainCamera == null || this == null)
```
Comment (AjimenezDCL) [inline unity-client/Assets/UnityGLTF/Scripts/GLTFComponent.cs:361]: This is mindblowing. Can `this` be `null`?

**Review by AjimenezDCL — APPROVED** [review]
Approving with minor requestes

**Pull request merged**